### PR TITLE
Use ExposureFitsReader.

### PR DESCRIPTION
**Pull request by ktlim**
Also corrects a bug that omitted a "repoInfo.", and allows search through parents with `getUri` unlike `calexp_filename`.

### CODE REVIEW

File: python/lsst/pipe/analysis/plotUtils.py [hunk 1]
```diff
@@ -485,27 +485,29 @@ def plotCcdOutline(axes, butler, dataId, ccdList, tractInfo=None, zpLabel=None,
...
-        w = calexp.getWidth()
-        h = calexp.getHeight()
+            wcs = butler.get("calexp_wcs", dataIdCopy)
+        bbox = butler.get("calexp_bbox", dataIdCopy)
+        w = bbox.getWidth()
+        h = bbox.getHeight()
         if zpLabel is not None:
             if hscRun and (zpLabel == "MEAS_MOSAIC" or "MEAS_MOSAIC_1" in zpLabel):
-                nQuarter = calexp.getDetector().getOrientation().getNQuarter()
+                nQuarter = reader.readDetector().getOrientation().getNQuarter()
```
Comment (timj) [inline python/lsst/pipe/analysis/plotUtils.py]: Where is `reader` defined? (is there no `_detector` component access?)

File: python/lsst/pipe/analysis/visitAnalysis.py [hunk 2]
```diff
@@ -550,19 +550,15 @@ def readCatalogs(self, dataRefList, dataset, repoInfo, aliasDictList=None, fakeC
...
-                # polygon
-                pix = lsstgeom.Box2I(lsstGeom.Point2I(0, 0),
-                                     lsstGeom.Point2I(0, 0))
-                exp = butler.get("calexp_sub", dataId, bbox=pix)
-                wcs = exp.getWcs()
-
-                # Hacky but fast way of getting just the mask
+                # getUri is less safe but enables us to use an efficient
+                # ExposureFitsReader
+                fname = repoInfo.butler.getUri("calexp", dataId)
```
Comment (timj) [inline python/lsst/pipe/analysis/visitAnalysis.py:555]: Why getUri rather than `_filename`?

**Review by timj — APPROVED** [review]
Looks okay. One issue and one question.

File: python/lsst/pipe/analysis/plotUtils.py (hunk 1 above)
Comment (laurenam) [inline python/lsst/pipe/analysis/plotUtils.py]: Does `reader` need to be defined before use here?

Comment (laurenam) [reply python/lsst/pipe/analysis/plotUtils.py]: Oh...tips Tim 😉

Comment (ktlim) [reply python/lsst/pipe/analysis/plotUtils.py]: It was from a previous iteration when I thought I needed it.  Thanks for the catch.

File: python/lsst/pipe/analysis/visitAnalysis.py (hunk 2 above)
Comment (ktlim) [reply python/lsst/pipe/analysis/visitAnalysis.py:555]: The former searches parents; the latter doesn't.

**Pull request merged**